### PR TITLE
develop: add ProxySQL integration guidance

### PR DESCRIPTION
**Pull request by Icemap**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v6.2 (TiDB 6.2 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: TOC.md [hunk 1]
```diff
@@ -80,6 +80,7 @@
     - [Gitpod](/develop/dev-guide-playground-gitpod.md)
   - 第三方软件支持
     - [PingCAP 维护的三方库](/develop/dev-guide-third-party-support.md)
+    - [TiDB 与 ProxySQL 集成方案](/develop/dev-guide-proxysql-integration.md)
```
Comment (Oreoxmt) [inline TOC.md:83]:
```suggestion
    - [TiDB 与 ProxySQL 集成](/develop/dev-guide-proxysql-integration.md)
```

File: develop/dev-guide-proxysql-integration.md [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+- `username`: 用户名
+- `password`: 密码
+- `active`: 是否生效，`1` 为生效，`0` 为不生效，仅 `active = 1` 的用户可登录。
+- `default_hostgroup`: 此账号默认使用的 **hostgroup**，SQL 将被发送至此 **hostgroup** 中，除非查询规则将流量发送到不同的 **hostgroup**。
+- `transaction_persistent`: 值为 `1` 时，表示事务持久化，即：当某连接使用该用户开启了一个事务后，那么在事务提交或回滚之前，
+所有的语句都路由到同一个 **hostgroup** 中，避免语句分散到不同 **hostgroup**。
+
+### 3.4 配置文件配置
+
+除了使用 **_ProxySQL Admin interface_** 配置，也可以使用配置文件进行配置。[官方解释](https://github.com/sysown/proxysql#configuring-proxysql-through-the-config-file)中，配置文件仅应该被视为是一种辅助初始化的方式，而并非主要配置的手段。配置文件仅在 SQLite 数据库未被创建时读取，后续将不会继续读取配置文件。因此，使用配置文件配置时，你应进行 SQLite 数据库的删除，这将**_丢失_**你在 **_ProxySQL Admin interface_** 中对配置进行的更改：
```
Comment (Mini256) [inline develop/dev-guide-proxysql-integration.md:151]: ** 挨着的话好像不会起作用，在 GitHub 上预览没有效果，不太确定文档站是不是也这样。

```suggestion
除了使用 **_ProxySQL Admin interface_** 配置，也可以使用配置文件进行配置。[官方解释](https://github.com/sysown/proxysql#configuring-proxysql-through-the-config-file)中，配置文件仅应该被视为是一种辅助初始化的方式，而并非主要配置的手段。配置文件仅在 SQLite 数据库未被创建时读取，后续将不会继续读取配置文件。因此，使用配置文件配置时，你应进行 SQLite 数据库的删除，这将 **_丢失_** 你在 **_ProxySQL Admin interface_** 中对配置进行的更改：
```

File: develop/dev-guide-proxysql-integration.md [hunk 3]
```diff
@@ -0,0 +1,486 @@
+---
+title: TiDB 与 ProxySQL 集成方案
+summary: 叙述 TiDB 与 ProxySQL 集成方案。
+---
+
+# TiDB 与 ProxySQL 集成方案
```
Comment (winkyao) [inline develop/dev-guide-proxysql-integration.md:6]:
```suggestion
# TiDB 与 ProxySQL 集成
```

File: develop/dev-guide-proxysql-integration.md [hunk 4]
```diff
@@ -0,0 +1,486 @@
...
+
+此文档以 CentOS 7 为例，简单介绍 **_TiDB_** 与 **_ProxySQL_** 的集成方法，如果你有其他系统的集成需求，可查看[快速体验](#4-快速体验)一节，这是使用 Docker 及 Docker Compose 部署测试集成环境的介绍。你也可以自行参阅以下链接，以获得更多信息：
+
+- [TiDB 文档](/overview.md)
+- [TiDB 开发者文档](/develop/dev-guide-overview.md)
+- [ProxySQL 官方文档](https://proxysql.com/documentation/)
+- [TiDB 与 ProxySQL 的集成测试](https://github.com/Icemap/tidb-proxysql-integration-test)
+
+## 1. 启动 TiDB
```
[the remark this answers is not in the document]
Comment (Icemap) [reply develop/dev-guide-proxysql-integration.md:15]: Done

File: develop/dev-guide-proxysql-integration.md (hunk 2 above)
Comment (Icemap) [reply develop/dev-guide-proxysql-integration.md:165]: 文档站是可以的，而且如果中文之间加了空格会不符合文档团队的规范😭😭😭

File: develop/dev-guide-proxysql-integration.md [hunk 5]
```diff
@@ -0,0 +1,486 @@
...
+### 3.5 其余配置项
+
+仅以上配置为必需配置项，其余配置项并非必需。你可在 ProxySQL 文档中的 [Global Variables](https://proxysql.com/documentation/global-variables/) 一文中获取全部配置项名称及其作用。
+
+## 4. 快速体验
+
+你可以使用 Docker 及 Docker Compose 快速进行集成后的环境体验，请确认 `4000`、`6033` 端口的空闲：
+
+```sh
+docker-compose up -d
```
[the remark this answers is not in the document]
Comment (Icemap) [reply develop/dev-guide-proxysql-integration.md:214]: Done

File: develop/dev-guide-proxysql-integration.md [hunk 6]
```diff
@@ -0,0 +1,486 @@
...
++--------------------+
+| 5.7.25-TiDB-v6.1.0 |
++--------------------+
+```
+
+### 5.2 运行集成测试
+
+如果你满足以下依赖项：
+
+- 测试用例代码仓库 [tidb-test](https://github.com/pingcap/tidb-test) 的权限
```
[the remark this answers is not in the document]
Comment (Icemap) [reply develop/dev-guide-proxysql-integration.md:243]: 是的，符合预期。

File: develop/dev-guide-proxysql-integration.md (hunk 2 above)
Comment (Mini256) [reply develop/dev-guide-proxysql-integration.md:165]: Got

File: develop/dev-guide-proxysql-integration.md [hunk 7]
```diff
@@ -0,0 +1,486 @@
...
+
+```sql
++--------------------+
+| VERSION()          |
++--------------------+
+| 5.7.25-TiDB-v6.1.0 |
++--------------------+
+```
+
+### 5.2 运行集成测试
```
[the remark this answers is not in the document]
Comment (Icemap) [reply develop/dev-guide-proxysql-integration.md]: Done

File: develop/dev-guide-proxysql-integration.md [hunk 8]
```diff
@@ -0,0 +1,486 @@
...
+    - `begin;insert into test.test (db) values ('insert this and rollback later'); select * from test.test; rollback;`: `insert` 语句预计不会匹配所有规则，因此将使用用户的 `default_hostgroup`（为 `0`），从而转发至 `hostgroup` 为 `0` 的 TiDB 后端 `tidb-0` 中。而因为 ProxySQL 默认开启用户的 `transaction_persistent`，这将使同一个事务内的所有语句运行在同一个 `hostgroup` 中，因此，这里的 `select * from test.test;` 也将转发至 `hostgroup` 为 `0` 的 TiDB 后端 `tidb-0` 中。
+6. 停止并清除 Docker Compose 启动的容器、网络拓扑等资源。
+
+#### 5.5.2 示例运行
+
+依赖：
+
+- Docker
+- Docker Compose
+- MySQL Client
```
[the remark this answers is not in the document]
Comment (Icemap) [reply develop/dev-guide-proxysql-integration.md:422]: Done

File: develop/dev-guide-proxysql-integration.md [hunk 9]
```diff
@@ -0,0 +1,486 @@
...
+        - Docker
+        - Docker Compose
+
+那么你可以克隆[集成测试代码](https://github.com/Icemap/tidb-proxysql-integration-test)并本地运行 `./proxysql-initial.sh && ./test-local.sh` 或使用 Docker 运行 `./test-docker.sh` 来运行集成测试。
+
+在[集成测试文档](https://github.com/Icemap/tidb-proxysql-integration-test/blob/main/README-zh.md)中有更多信息可供查看。
+
+### 5.3 配置负载均衡示例 - 使用 Admin Interface 进行配置
+
+#### 5.3.1 示例操作步骤
```
[the remark this answers is not in the document]
Comment (Icemap) [reply develop/dev-guide-proxysql-integration.md]: Done

File: develop/dev-guide-proxysql-integration.md [hunk 10]
```diff
@@ -0,0 +1,637 @@
...
+## 3. 配置 ProxySQL
+
+我们需要将 ProxySQL 内的配置指向 TiDB，以此将 ProxySQL 作为 TiDB 的代理。此处将列举必需的配置项，其余配置项可参考 ProxySQL [官方文档](https://proxysql.com/documentation/)。
+
+### 3.1 ProxySQL 配置的简单介绍
+
+ProxySQL 使用一个单独的端口进行配置管理，另一个端口进行代理。我们把配置管理的入口称为 **_ProxySQL Admin interface_**，把代理的入口称为 **_ProxySQL MySQL interface_**。
+
+- **_ProxySQL Admin interface_**: 可以使用具有 `admin` 权限的用户连接到管理界面，以读取和写入配置，或者使用具有 `stats` 权限的用户，只能读取某些统计数据（不读取或写入配置）。默认凭证是 `admin:admin` 和 `stats:stats`，但出于安全考虑，可以使用默认凭证进行本地连接。要远程连接，需要配置一个新的用户，通常它被命名为 `radmin`。
+- **_ProxySQL MySQL interface_**: 用于代理，将 SQL 转发到配置的服务中。
```
Comment (Mini256) [inline develop/dev-guide-proxysql-integration.md:120]: "ProxySQL MySQL Interface" 统一大小写

File: develop/dev-guide-proxysql-integration.md [hunk 11]
```diff
@@ -0,0 +1,637 @@
...
+mysql_users:
+ (
+    {
+         username = "root"
+        password = ""
+         default_hostgroup = 0
+         max_connections = 1000
+         default_schema = "test"
+         active = 1
+        transaction_persistent = 1
```
Comment (Mini256) [inline develop/dev-guide-proxysql-integration.md:195]: 对齐

**Review by Mini256 — APPROVED** [review]
rest LGTM

File: develop/dev-guide-proxysql-integration.md [hunk 12]
```diff
@@ -0,0 +1,639 @@
...
+2. 随后，你可以使用配置文件 [`tidb-config.toml`](https://github.com/Icemap/tidb-proxysql-integration-test/blob/main/tidb-config.toml) 来启动 TiDB，命令如下所示：
+
+    ```shell
+    ${TIDB_SERVER_PATH} -config ./tidb-config.toml -store unistore -path "" -lease 0s > ${LOCAL_TIDB_LOG} 2>&1 &
+    ```
+
+    > **注意：**
+    >
+    > - 此处使用 `unistore` 作为存储引擎，这是 TiDB 的测试存储引擎，请仅在测试时使用它。
+    > - `TIDB_SERVER_PATH`：上一步中使用 `go build` 编译的二进制文件位置，如你在 `/usr/local` 下进行上一步操作，那么此处的 `TIDB_SERVER_PATH` 应为：`/usr/local/tidb/tidb-server/tidb-server`。
```
Comment (Oreoxmt) [inline develop/dev-guide-proxysql-integration.md:40]:
```suggestion
    > - `TIDB_SERVER_PATH`：上一步中使用 `go build` 编译的二进制文件位置，如你在 `/usr/local` 下进行上一步操作，那么此处的 `TIDB_SERVER_PATH` 应为 `/usr/local/tidb/tidb-server/tidb-server`。
```

File: develop/dev-guide-proxysql-integration.md [hunk 13]
```diff
@@ -0,0 +1,639 @@
...
++--------------------+
+```
+
+## 5. 配置示例
+
+示例依赖：
+
+- Docker
+- Docker Compose
+- MySQL Client
```
Comment (Oreoxmt) [inline develop/dev-guide-proxysql-integration.md:251]: 这些依赖是否有版本要求

Comment (Icemap) [reply develop/dev-guide-proxysql-integration.md]: 无，我服务器上使用的是非常老版本的 Docker Compose，无需注释了。

**Review by Oreoxmt — APPROVED** [review]
LGTM

File: develop/dev-guide-proxysql-integration.md [hunk 14]
```diff
@@ -0,0 +1,656 @@
...
+
+1. 下载 [TiDB](https://github.com/pingcap/tidb) 源码，进入 `tidb-server` 目录后，使用 `go build` 进行编译。
+
+    ```shell
+    git clone git@github.com:pingcap/tidb.git
+    cd tidb/tidb-server
+    go build
+    ```
+
+2. 随后，你可以使用配置文件 [`tidb-config.toml`](https://github.com/Icemap/tidb-proxysql-integration-test/blob/main/tidb-config.toml) 来启动 TiDB，命令如下所示：
```
Comment (qiancai) [inline develop/dev-guide-proxysql-integration.md:31]:
```suggestion
2. 使用配置文件 [`tidb-config.toml`](https://github.com/Icemap/tidb-proxysql-integration-test/blob/main/tidb-config.toml) 来启动 TiDB，命令如下所示：
```

File: develop/dev-guide-proxysql-integration.md [hunk 15]
```diff
@@ -0,0 +1,656 @@
...
+
+### 其他安装方式
+
+参考 ProxySQL 的 [Github 页面](https://github.com/sysown/proxysql#installation)或 [ProxySQL 官方文档](https://proxysql.com/documentation/)进行安装。
+
+## 3. 配置 ProxySQL
+
+需要将 ProxySQL 内的配置指向 TiDB，以此将 ProxySQL 作为 TiDB 的代理。下面列举必需的配置项，其余配置项可参考 [ProxySQL 官方文档](https://proxysql.com/documentation/)。
+
+### ProxySQL 配置的简单介绍
```
Comment (qiancai) [inline develop/dev-guide-proxysql-integration.md:125]:
```suggestion
### ProxySQL 配置简介
```

File: develop/dev-guide-proxysql-integration.md [hunk 16]
```diff
@@ -0,0 +1,656 @@
...
+
+ProxySQL 使用一个单独的端口进行配置管理，另一个端口进行代理。我们把配置管理的入口称为 **_ProxySQL Admin interface_**，把代理的入口称为 **_ProxySQL MySQL Interface_**。
```
Comment (qiancai) [inline develop/dev-guide-proxysql-integration.md:127]:
```suggestion
ProxySQL 使用一个单独的端口进行配置管理，另一个端口进行代理。其中，配置管理的入口称为 **_ProxySQL Admin interface_**，代理的入口称为 **_ProxySQL MySQL Interface_**。
```

File: develop/dev-guide-proxysql-integration.md [hunk 17]
```diff
@@ -0,0 +1,656 @@
...
+
+除了使用 **_ProxySQL Admin interface_** 配置，也可以使用配置文件进行配置。[ProxySQL 文档](https://github.com/sysown/proxysql#configuring-proxysql-through-the-config-file)中，配置文件仅应该被视为是一种辅助初始化的方式，而并非主要配置的手段。配置文件仅在 SQLite 数据库未被创建时读取，后续将不会继续读取配置文件。因此，使用配置文件配置时，你应进行 SQLite 数据库的删除，这将**丢失**你在 **_ProxySQL Admin interface_** 中对配置进行的更改：
+
+```shell
+rm /var/lib/proxysql/proxysql.db
+```
+
+另外，也可以运行 `LOAD xxx FROM CONFIG`，用配置文件中的配置覆盖当前内存中的配置。
+
+配置文件的位置为 `/etc/proxysql.cnf`，我们将上方的必需配置翻译为配置文件方式，仅更改 `mysql_servers`、`mysql_users` 这两个配置节点，其余配置可自行查看 `/etc/proxysql.cnf`：
```
Comment (qiancai) [inline develop/dev-guide-proxysql-integration.md:182]: “翻译为”，是否可以找一个词替换呢？

File: develop/dev-guide-proxysql-integration.md [hunk 18]
```diff
@@ -0,0 +1,656 @@
...
+
+随后使用 `systemctl restart proxysql` 进行服务重启后即可生效，配置生效后将自动创建 SQLite 数据库，后续将不会再次读取配置文件。
+
+### 其余配置项
+
+仅以上配置为必需配置项，其余配置项并非必需。你可在 [ProxySQL Global Variables](https://proxysql.com/documentation/global-variables/) 中获取全部配置项的名称及作用。
+
+## 4. 快速体验
+
+你可以使用 Docker 及 Docker Compose 快速进行集成后的环境体验，请确认 `4000`、`6033` 端口未被占用，然后执行如下命令：
```
Comment (qiancai) [inline develop/dev-guide-proxysql-integration.md:217]:
```suggestion
在测试环境中，你可以使用 Docker 及 Docker Compose 快速进行集成后的环境体验，请确认 `4000`、`6033` 端口未被占用，然后执行如下命令：
```

File: develop/dev-guide-proxysql-integration.md [hunk 19]
```diff
@@ -0,0 +1,656 @@
...
+
+#### 逐步运行
+
+1. 通过 Docker Compose 启动三个 TiDB 容器实例，容器内部端口均为 `4000`，映射宿主机端口为 `4001`、`4002`、`4003`。TiDB 容器实例启动后，再启动一个 ProxySQL 实例，容器内部 **_ProxySQL MySQL Interface_** 端口为 `6033`，映射宿主机端口为 `6034`。不暴露 **_ProxySQL Admin Interface_** 端口，因为其仅可在本地（即容器内）登录 **_ProxySQL Admin Interface_**。此流程被写在 [`docker-compose.yaml`](https://github.com/Icemap/tidb-proxysql-integration-test/blob/main/example/load-balance-admin-interface/docker-compose.yaml) 中。
+
+    ```shell
+    docker-compose up -d
+    ```
+
+2. 在 3 个 TiDB 实例内，创建相同的表结构，但写入不同的数据：`'tidb-0'`、`'tidb-1'`、`'tidb-2'`，以便分辨不同的数据库实例：
```
Comment (qiancai) [inline develop/dev-guide-proxysql-integration.md:285]: 是否需要 1. 启动 TiDB 比如创建的 TiDB 集群包含 3 个节点呢？。如果是，前面可以说下吗

File: develop/dev-guide-proxysql-integration.md [hunk 20]
```diff
@@ -0,0 +1,656 @@
...
+你可以使用 Docker 及 Docker Compose 快速进行集成后的环境体验，请确认 `4000`、`6033` 端口未被占用，然后执行如下命令：
+
+```shell
+git clone https://github.com/Icemap/tidb-proxysql-integration-test.git
+cd tidb-proxysql-integration-test && docker-compose pull # Get the latest Docker images
+sudo setenforce 0 # Only on Linux
+docker-compose up -d
+```
+
+这样就已经完成了一个集成了 TiDB 与 ProxySQL 环境的启动，这将启动两个容器。**请勿**在生产环境使用此快速体验方式创建集成环境。你可以使用用户名为 `root`，密码为空的账号，登录到本机的 `6033` 端口 (ProxySQL)。容器具体配置可见 [`docker-compose.yaml`](https://github.com/Icemap/tidb-proxysql-integration-test/blob/main/docker-compose.yaml)，ProxySQL 具体配置可见 [proxysql-docker.cnf](https://github.com/Icemap/tidb-proxysql-integration-test/blob/main/proxysql-docker.cnf)。
```
Comment (qiancai) [inline develop/dev-guide-proxysql-integration.md:226]: 生产环境如果想要体验创建集群，可以参考什么文档呢

File: develop/dev-guide-proxysql-integration.md [hunk 21]
```diff
@@ -0,0 +1,660 @@
...
+./test-load-balance.sh
+```
+
+此配置实现效果与[使用 Admin Interface 配置负载均衡](#使用-admin-interface-配置负载均衡)完全一致，仅改为使用配置文件进行 ProxySQL 初始化配置。
+
+> **注意：**
+>
+> - ProxySQL 的配置保存在 SQLite 中。配置文件仅在 SQLite 不存在时读取。
+> - ProxySQL **不建议**使用配置文件进行配置更改，仅作为初始化配置时使用，请勿过度依赖配置文件。
+> - 这是由于使用 **_ProxySQL Admin Interface_** 配置时，会有以下优点：
```
Comment (qiancai) [inline develop/dev-guide-proxysql-integration.md:655]:
```suggestion
> - ProxySQL **不建议**使用配置文件进行配置更改，仅作为初始化配置时使用，请勿过度依赖配置文件。这是由于使用 **_ProxySQL Admin Interface_** 配置时，会有以下优点：
```

File: develop/dev-guide-proxysql-integration.md (hunk 20 above)
Comment (Icemap) [reply develop/dev-guide-proxysql-integration.md]: 上方有写如何在生产环境创建集群。

File: develop/dev-guide-proxysql-integration.md (hunk 19 above)
Comment (Icemap) [reply develop/dev-guide-proxysql-integration.md:289]: 这里有啊：https://github.com/pingcap/docs-cn/pull/10706/files#diff-47a484d02ef3f9c897fdfe68d11377cf0a8ab12998d92b84d13d400ae52030d9R283

File: develop/dev-guide-proxysql-integration.md [hunk 22]
```diff
@@ -0,0 +1,659 @@
...
+在测试环境中，你可以使用 Docker 及 Docker Compose 快速进行集成后的环境体验，请确认 `4000`、`6033` 端口未被占用，然后执行如下命令：
+
+```shell
+git clone https://github.com/Icemap/tidb-proxysql-integration-test.git
+cd tidb-proxysql-integration-test && docker-compose pull # Get the latest Docker images
+sudo setenforce 0 # Only on Linux
+docker-compose up -d
+```
+
+> **注意：**
```
Comment (qiancai) [inline develop/dev-guide-proxysql-integration.md:226]:
```suggestion
> **警告：**
```

**Review by qiancai — APPROVED** [review]
LGTM

**Pull request merged**